public/scripts/index.js

```diff
@@ -1,138 +1,162 @@
+function getInputValue () {
+  /* const rows = document.getElementById('rows').value
+  const cols = document.getElementById('columns').value
 
-    const h = rows * cellSize;
-    canvas.height = h;
+  const h = rows * cellSize
+  canvas.height = h
 
-    const w = cols * cellSize;
-    canvas.width = w;
+  const w = cols * cellSize
+  canvas.width = w
 
-    const arrH = h / cellSize;
-    const arrW = w / cellSize;
+  const arrH = h / cellSize
+  const arrW = w / cellSize
 
-    currentFrame = makeMatrix(arrH, arrW);
-    nextFrame = makeMatrix(arrH, arrW);
-    intermediateFrame = makeMatrix(arrH, arrW);
+  currentFrame = makeMatrix(arrH, arrW)
+  nextFrame = makeMatrix(arrH, arrW)
+  intermediateFrame = makeMatrix(arrH, arrW)
+  // drawBoard(w, h, context, cellSize) */
+}
 
-    let context = canvas.getContext("2d");
+function simulate () {
+  let Nwid = (Math.ceil(window.innerWidth / 8) * 8)
+  let Nhei = (Math.ceil(window.innerHeight / 8) * 8)
 
-    drawBoard(w, h, context, cellSize);
-}
+  let wid = Nwid - 32
+  let hei = Nhei - 32
 
-function simulate(){
-    let rows = document.getElementById("rows").value;
-    let cols = document.getElementById("columns").value;
+  let rows = hei / cellSize
+  let cols = wid / cellSize
 
-    let intermediateFrame = getIntermediateFrame(currentFrame, rows, cols);
-    for (let i = 0; i < intermediateFrame.length; i++){
-        for (let j = 0; j < intermediateFrame[0].length; j++){
-                nextFrame[i][j] = calcNewCellState(currentFrame[i][j], intermediateFrame[i][j]);
-        }
+  const intermediateFrame = getIntermediateFrame(currentFrame, rows, cols)
+  for (let i = 0; i < intermediateFrame.length; i++) {
+    for (let j = 0; j < intermediateFrame[0].length; j++) {
+      nextFrame[i][j] = calcNewCellState(currentFrame[i][j], intermediateFrame[i][j])
     }
-    currentFrame = nextFrame;
-    draw(currentFrame, context, cellSize);        
+  }
+  currentFrame = nextFrame
+  draw(currentFrame, context, cellSize)
 }
 
-function drawPreset(){
-    let value = getPreset();
-    let numValue = parseInt(value);
-    if (numValue === 1 && currentFrame.length >= 17 && currentFrame[16][16] != undefined ){
-        clear();
-        drawToad(currentFrame, context, cellSize);
-    }
-    if (numValue === 2 && currentFrame.length >= 17 && currentFrame[16][16] != undefined){
-        clear();
-        drawAcorn(currentFrame, context, cellSize);
-    }
-    if (numValue === 3 && currentFrame.length >= 17 && currentFrame[16][16] != undefined){
-        clear();
-        drawLWSS(currentFrame, context, cellSize)
-    }
-
-    if (currentFrame === undefined){
-        alert('you must draw a grid')
-    }
-    else if (numValue === 0){
-        alert('You must select a preset');
-    }
-    else if (currentFrame.length < 17){
-        alert('Grid is too small. The minimum size is 17x17');
-    }    
+function drawPreset () {
+  let Nwid = (Math.ceil(window.innerWidth / 8) * 8)
```

Comment:
instead of recalculating `Nwid` and `Nhei` in each func you should calc it once and save the value in a module scope variable.